WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt

```diff
@@ -88,38 +87,14 @@ class ReaderDiscoverViewModel @Inject constructor(
 
         // Listen to changes to the discover feed
         _uiState.addSource(readerDiscoverDataProvider.discoverFeed) { posts ->
-            if (posts != null && posts.cards.isNotEmpty()) {
-                _uiState.value = ContentUiState(
-                        posts.cards.map {
-                            when (it) {
-                                is ReaderPostCard -> postUiStateBuilder.mapPostToUiState(
-                                        post = it.post,
-                                        isDiscover = true,
-                                        photonWidth = photonWidth,
-                                        photonHeight = photonHeight,
-                                        isBookmarkList = false,
-                                        onButtonClicked = this::onButtonClicked,
-                                        onItemClicked = this::onPostItemClicked,
-                                        onItemRendered = this::onItemRendered,
-                                        onDiscoverSectionClicked = this::onDiscoverClicked,
-                                        onMoreButtonClicked = this::onMoreButtonClicked,
-                                        onMoreDismissed = this::onMoreMenuDismissed,
-                                        onVideoOverlayClicked = this::onVideoOverlayClicked,
-                                        onPostHeaderViewClicked = this::onPostHeaderClicked,
-                                        onTagItemClicked = this::onTagItemClicked,
-                                        postListType = TAG_FOLLOWED
-                                )
-                                is InterestsYouMayLikeCard -> {
-                                    postUiStateBuilder.mapTagListToReaderInterestUiState(
-                                            it.interests,
-                                            this::onReaderTagClicked
-                                    )
-                                }
-                            }
-                        },
-                        reloadProgressVisibility = false,
-                        loadMoreProgressVisibility = false
-                )
+            launch {
```

Comment:
I've moved the body into `convertCardsToUiStates` and we are launching a coroutine here now.